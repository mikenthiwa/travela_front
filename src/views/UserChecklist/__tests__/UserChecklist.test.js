import React from 'react';
import { mount } from 'enzyme';
import {UserChecklist} from '..';

describe ('UserChecklist Test Suite', () => {
  const props = {
    id: 'jdjjfkfk',
    isLoading: false,
    fullName: 'Jude Afam',
    checklists: [{
      id: 'jjdd',
      config: []
    },{
      id: 'jjjk',
      config: []
    }
    ],
    trips: [],
    isSaving: false,
    closeModal: jest.fn(),
    isSubmitted: false,
    completionCount: 100,
    location: {
      pathname: '/new-requests/4WD6f5YaJ/checklists'
    },
    fetchChecklistSubmission: jest.fn(),
    updateChecklistSubmission: jest.fn(),
    postChecklistSubmission: jest.fn(),
    openModal: jest.fn(),
    shouldOpen: false
  };

  const wrapper = mount(<UserChecklist {...props} />);
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.length).toBe(1);
  });

  it('should submit checklist', () => {
    wrapper.instance().handleSubmitChecklist();
    wrapper.instance().postSubmission();
    expect(props.closeModal).toBeCalled();
  });

  it('should update checklist', () => {
    wrapper.instance().updateChecklistSubmission();
    expect(props.updateChecklistSubmission).toBeCalled();
  });

  it('should click next button and open submit modal', () => {
    wrapper.instance().onNextButtonClick();
    wrapper.instance().onTabChange(2);
    wrapper.find('.bg-btn').at(1).simulate('click');
    expect(props.openModal).toHaveBeenCalled();
  });

  it('should click the back button', () => {
    wrapper.find('.bg-btn').first().simulate('click');
    wrapper.instance().onTabChange(1);
    expect(wrapper.state('tabIndex')).toBe(1);
  });
});
