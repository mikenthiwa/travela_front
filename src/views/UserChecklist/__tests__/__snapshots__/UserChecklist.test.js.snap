// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserChecklist Test Suite should render correctly 1`] = `
Array [
  "",
  <div
    className="smart-checklist-wrapper"
>
    <div
        className="smart-checklist"
    >
        <div
            className="smart-checklist-header"
        >
            <div
                className="details"
            >
                <p>
                    Your Travel Checklist
                </p>
                Hi Jude kindly fill in your details to complete your travel checklist
            </div>
            <div
                className="circular-calculator"
            >
                <svg
                    height={50}
                    viewBox="0 0 50 50"
                    width={50}
                >
                    <circle
                        className="circle-background"
                        cx={25}
                        cy={25}
                        r={22.5}
                        strokeWidth="5px"
                    />
                    <circle
                        className="circle-progress"
                        cx={25}
                        cy={25}
                        r={22.5}
                        stroke="#10A36D"
                        strokeWidth="5px"
                        style={
                            Object {
                                "strokeDasharray": 141.3716694115407,
                                "strokeDashoffset": 0,
                              }
                        }
                        transform="rotate(-90 25 25)"
                    />
                    <text
                        className="circle-text"
                        dy=".3em"
                        textAnchor="middle"
                        x="50%"
                        y="50%"
                    >
                        100%
                    </text>
                </svg>
            </div>
        </div>
        <div
            className="smart-checklist-body"
        >
            <undefined />
        </div>
    </div>
    <div
        className="smart-checklist-tab-buttons"
    >
        <button
            className="bg-btn bg-btn--active"
            disabled={true}
            onClick={[Function]}
            type="button"
        >
            Back
        </button>
        <button
            className="bg-btn bg-btn--active"
            disabled={false}
            onClick={[Function]}
            type="button"
        >
            Next
        </button>
    </div>
</div>,
]
`;
